WooCommerce/Classes/ViewModels/InAppPurchases/UpgradesViewModel.swift

```diff
@@ -30,8 +30,9 @@ enum PrePurchaseError: Error {
 }
 
 enum PurchaseUpgradeError {
-    case inAppPurchaseFailed(WooWPComPlan)
-    case planActivationFailed
+    case inAppPurchaseFailed(WooWPComPlan, InAppPurchaseStore.Errors)
```

Comment:
TIL we can pass more than one associated type 💯  

Not now, but I'm wondering if we should think of refactoring the different enums here in some sort of Result type like: `InAppPurchaseResult<WooWPComPlan, InAppPurchaseStore.Error>`, then update the `UpgradeViewState` based on the result case we get back.

Reply:
Maybe? We won't always have a "result" per se though. The view state declares all the... states the view can be in. 

Not all of those states would fit well in a `Result`, since several are intermediate states. Having all the view states in a single enum means we can switch over it in one place and be sure that we're covering all the cases, without any confusion over shared state.

Reply:
In this case, we need both the `WooWPComPlan` and the error, and a result would only give you one of them.